Fix: categorized Extensibility section and sync navigation

zihanKuang opened the pull request:
**Notes for Reviewers**

- This PR fixes #14113 

- [x] Yes, I signed my commits.

This PR improves theExtensibility section in Meshery’s documentation and corrects the navigation structure:

#### Enhancements:
-  Categorized content under the Extensibility section for improved clarity and easier navigation.
-  Refactored `extensibility.md` to `index.md` for a cleaner URL structure (`/extensibility/`).
-  Added proper redirects to prevent 404 errors for existing links.

#### Navigation Sync:
-  Synchronized navigation across:
  - `_data/toc.yml` (documentation sidebar structure),
  - `_includes/sidebar.html` (dynamic sidebar rendering),
  - `pages/index.md` (homepage navigation display).
-  Fixed incorrect nesting in the navigation (`children` → `grandchildren`), which caused certain entries to not display.
-  Aligned the homepage navigation with the updated `toc.yml` for consistency.
-  Added an **Extensibility section** to the homepage for complete coverage and easier access.


#### Open Questions

1. **Emoji Consistency in Section Titles**  
   Currently, emojis are used inconsistently in section titles (e.g., some sections on the homepage have emojis, others don’t).  
   → Should we standardize emoji usage across all section titles or only highlight major sections (e.g., homepage navigation)?

2. **Placement of the Extensibility Section**  
   The optimal location for the **Extensibility** section is under consideration.  
   → Should it remain as a separate section on the homepage or be merged with **Integrations & Extensions** for better cohesion?

#### Next Steps

Update the **Documentation Contribution Guidelines** to instruct contributors to:
 Check and update both the navigation bar (`toc.yml`) and homepage (`index.md`) when adding or editing documentation pages.

l5io commented:
🚀 Preview for commit 51dc8490ec84ea93d47093d40f9216efe2f6a541 at: https://67e6f86a1f15ae0069985d7a--meshery-docs-preview.netlify.app

leecalcote commented:
Emojis: only highlighting major sections makes sense to me. I think the Tutorials was highlighted as a special case to help users get up and going.

zihanKuang commented:
[like]  Zihan Kuang reacted to your message:
________________________________
From: Lee Calcote ***@***.***>
Sent: Friday, March 28, 2025 7:47:10 PM
To: meshery/meshery ***@***.***>
Cc: Zihan Kuang ***@***.***>; Author ***@***.***>
Subject: Re: [meshery/meshery] Fix: categorized Extensibility section and sync navigation (PR #14116)


Emojis: only highlighting major sections makes sense to me. I think the Tutorials was highlighted as a special case to help users get up and going.

—
Reply to this email directly, view it on GitHub<https://github.com/meshery/meshery/pull/14116#issuecomment-2762283278>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A6JRDZTWYEDQO4DB3PJPC5L2WWRL5AVCNFSM6AAAAABZVTNBEGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONRSGI4DGMRXHA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

[leecalcote]leecalcote left a comment (meshery/meshery#14116)<https://github.com/meshery/meshery/pull/14116#issuecomment-2762283278>

Emojis: only highlighting major sections makes sense to me. I think the Tutorials was highlighted as a special case to help users get up and going.

—
Reply to this email directly, view it on GitHub<https://github.com/meshery/meshery/pull/14116#issuecomment-2762283278>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A6JRDZTWYEDQO4DB3PJPC5L2WWRL5AVCNFSM6AAAAABZVTNBEGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONRSGI4DGMRXHA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

l5io commented:
🚀 Preview for commit 4ab0fdd4852f9616fb87b1dca03f5879c9d9f962 at: https://67e707c836c05b0fc9e37e83--meshery-docs-preview.netlify.app

leecalcote commented on docs/pages/extensibility/index.md:
```diff
@@ -1,12 +1,14 @@
 layout: default
 title: Extensibility
-permalink: extensibility
+permalink: extensibility/
```
I believe that this `/` needs to be removed.

leecalcote commented on docs/pages/index.md:
```diff
@@ -280,25 +280,34 @@ list: exclude
+  <details>
+    <summary>
+      <p style="display:inline">
+        <a href="{{ site.baseurl }}/extensibility/index" class="text-black">Extensibility</a>
```
`/index` doesn't work.

l5io commented:
🚀 Preview for commit f5472f22fd09daa7c049fd74a62c3e37a33255ad at: https://67e7e6bee0266d2278c73614--meshery-docs-preview.netlify.app

l5io commented:
🚀 Preview for commit 010d98acf7a5786889b47bd05505e09127a34da4 at: https://67e7eb1e06dcdce91d4ff379--meshery-docs-preview.netlify.app

zihanKuang commented:
HI @leecalcote @miacycle 
I’ve made several updates regarding the **Extensibility** section:

1. Fixed the broken link to the **Extensibility index**.  
2. Removed **Extensibility** from the TOC and placed it under the **Reference** section.  
3. Added a collapsible panel for **Extensibility** on the Home page under the **Reference** section.

That said, there are still a couple of issues🫤:

1. In the TOC, 🛠 **Integrations & Extensions** isn’t consistent with the other top-level sections — it’s missing the bullet point in front. I tried to fix it but couldn't get it to behave like the others. I think it’s because the structure is different: instead of being a parent item with nested children, it’s just two standalone links grouped together. Also, there’s no index page for the whole section, so the TOC doesn’t treat it as a proper category.
I’m thinking we could either create a new `Integrations & Extensions/index.md` page to give it a proper parent structure, or just leave it as-is for now and revisit it when we do a broader content restructuring.

3. When I preview the site, clicking on **Extensibility** unexpectedly redirects to the official website. It works fine locally, so I’m not sure if this is just on my end or a deployment-related issue.🤔

vr-varad commented:
I think the second issue is with the redirect_from
try wth 
https://67e7e6bee0266d2278c73614--meshery-docs-preview.netlify.app/extensibility/integrations
https://67e7e6bee0266d2278c73614--meshery-docs-preview.netlify.app/extensibility/extensions
https://67e7e6bee0266d2278c73614--meshery-docs-preview.netlify.app/extensibility
all redirect to main website @zihanKuang

l5io commented:
🚀 Preview for commit b0eda57bc41b7111c63e7b82d23fbbac7206a210 at: https://67e8fda56073b1535eed2f7e--meshery-docs-preview.netlify.app

vr-varad commented:
for the first issue I think something is up with the url
![Screenshot from 2025-03-30 19-12-59](https://github.com/user-attachments/assets/801f5333-f55d-484e-840a-5d245e7bc4c8)
![Screenshot from 2025-03-30 19-13-08](https://github.com/user-attachments/assets/d1ac3bb7-4d69-4721-817a-3e91e4bf7cd9)
I changed the url and bullet points are showing.
@zihanKuang

zihanKuang commented:
Thanks for checking, @vr-varad!

I actually mentioned the same thing earlier in the PR comments — I suspected that the absence of an `index.md` file for **Integrations & Extensions** is why the TOC doesn’t recognize it as a proper parent item, which results in the missing bullet point. Right now, the section is composed of two sibling links grouped together, rather than following a parent-child structure like the others.

As I see it, there are still two options:  
1️⃣ We could create a new `Integrations & Extensions/index.md` page to establish a clear parent structure.  
2️⃣ Or we could leave it as-is for now and revisit the structure during a broader content reorganization.

l5io commented:
🚀 Preview for commit a214c0e7ded9ff4d6e461cc6dc8e749914ba1ef1 at: https://67f006f68c9f995f2dd0ee00--meshery-docs-preview.netlify.app

leecalcote commented on docs/_data/toc.yml:
```diff
@@ -113,196 +119,199 @@
+        - title: CLI Authentication
           url: guides/mesheryctl/authenticate-with-meshery-via-cli
-        - title: Running system checks using Meshery CLI
+        - title: System Checks
```
@zihanKuang, is the doc type here a "task"? 

The prior (current) phrasing used for the page title is based on that notion.

l5io commented:
🚀 Preview for commit e08022501291d79af1c2fb7214f2de11ace49dcf at: https://67f013ccae605600b26d60d1--meshery-docs-preview.netlify.app

leecalcote commented on docs/_data/toc.yml:
```diff
@@ -113,196 +119,199 @@
 #   links:
 - title: 🛠 Integrations & Extensions
-  url: extensibility
+  url: "#"
```
@zihanKuang, I wonder if we intend to have a category landing page here akin to that of pages like https://docs.meshery.io/guides

zihanKuang replied:
Yes, I think we do need a landing page like that.

leecalcote commented on docs/_data/toc.yml:
```diff
@@ -113,196 +119,199 @@
-      url: guides/infrastructure-management/gitops-with-meshery
     - title: Performance Management ↆ
-      url: guides/performance-management
+      url: guides/performance-management/performance-management
```
Shouldn't this URL remain as-is, pointing to the section index? https://docs.meshery.io/guides/performance-management

zihanKuang replied:
I’ve linked the section titles to the same-named overview pages (e.g., `infrastructure-management.md`) in cases where both an `index.md` and overview exist. These pages already provide a full intro to the section, so i use them as the landing point.

leecalcote replied:
I hear that. 

Imagine for a moment that you are reading a book. You open the page to Chapter 2 and there is a chapter outline offered. You read for a bit. You open the page to Chapter 3 and there is a chapter outline offered. You read for a bit. You open the page to Chapter 4 and there is NO chapter outline offered, the chapter content just starts instead. You read for a bit. You open the page to Chapter 5 and there is a chapter outline offered. You read for a bit. 

I wonder if you might consider your Chapter 4 experience slightly jarring, perhaps, not the best reader experience ("UX"), and if you might consider the missing chapter outline an error on the publisher's part.

zihanKuang replied:
That makes sense. I’ll update the sections to keep the index.md as the landing page and list the “Overview” pages explicitly as children.

leecalcote replied:
It the symmetry, the uniformity that is of concern. Consistency of style improves readability.

zihanKuang replied:
Got it. I’ll make sure to follow this principle in future updates.

leecalcote commented on docs/_data/toc.yml:
```diff
@@ -113,196 +119,199 @@
       url: guides/troubleshooting
       children:
-        - title: Installation
+        - title: Installation Issues
```
This feels asymmetric. If "Issues" is added here, it seems like "Issues" would be added to the other page titles.

leecalcote commented on docs/_data/toc.yml:
```diff
@@ -113,196 +119,199 @@
+        - title: Operator & MeshSync
           url: guides/troubleshooting/meshery-operator-meshsync
-        - title: Meshery Server
+        - title: Runtime Errors
```
Does "Runtime" include or exclude Meshery Operator and MeshSync?

As a reader, browsing the Troubleshooting section, I can't say that I know what topic to expect being "Runtime Errors", while I do know what topic to expect behind "Meshery Server".

leecalcote commented on docs/_data/toc.yml:
```diff
@@ -113,196 +119,199 @@
   url: project
   links:
-    - title: Community
+    - title: Meshery Community
```
How do you reconcile the justification for removing "...for Meshery" from the title of tutorials against the justification for the need to specify that we're talking about the "Meshery" Community?

I understand removing "...for Meshery" from the tutorial titles given that the topic of Meshery is implicitly understood. Adding "Meshery" to the title of Community is helpful because we need to distinguish between communities? If that isn't the case, then the lack of uniformity for inclusion and exclusion of such specificity seems to be unevenly applied. In applying the same test to the title of each page, I advocate for leaving "Community" as-is and shortening the tutorial titles.

leecalcote commented on docs/_data/toc.yml:
```diff
@@ -113,196 +119,199 @@
+        - title: Contributing to Adapters
           url: project/contributing/contributing-adapters
-        - title: CLI ↆ
+        - title: CLI Contributions ↆ
```
Stylistically, if our choice is to apply redundant wording to the section titles, why then, don't each of the Troubleshooting pages include "Troubleshooting" in there name (e.g. "Meshery Server Troubleshooting")?

I suggest that we leave "Contributions" out.

l5io commented:
🚀 Preview for commit 376810608890ecde3b587d76f10fb31058473636 at: https://67f060dbaf441634a6b200cc--meshery-docs-preview.netlify.app

zihanKuang commented:
@leecalcote 
I'm back with the updates!
Here’s what I’ve done in this round of changes:

1. For sections that have an `index.md`, I’ve restored them as the landing pages and added their corresponding "Overview" pages as children.
2. Removed redundant words "Contributions" and "Reference" from section titles for consistency.
3. While editing, I found a bug in the `/guides/` section where the child links on the `docs/pages/guides/index.md` page were incorrect.Turns out the TOC, home page, and section index pages each had their own separate hardcoded links 🫤 — which probably explains why I missed the broken ones earlier.
4. Also fixed the broken links in home page `docs/pages/index.md`:  
   - Updated the “Overview & Installation” link to point to the install index  
   - Fixed the incorrect link for “Reference”

leecalcote commented:
Great notes. These each make sense. 👍

leecalcote commented on docs/_data/toc.yml:
```diff
@@ -102,59 +102,71 @@
             - title: Registering a Connection
               url: guides/infrastructure-management/registering-a-connection
-    - title: Configuring Kubernetes Infrastructure ↆ
+    - title: Configuring Management ↆ
```
```suggestion
    - title: Configuration Management ↆ
```
